live-build/misc/live-build-hooks/81-upgrade-repository.binary

```diff
@@ -26,6 +26,18 @@
 
 set -o pipefail
 
+#
+# Sanity check to make sure we haven't installed multiple kernel versions.
+# This could happen if the version of the ZFS modules we install is not the
+# same as the kernel version installed with the package 'linux-generic'.
```

Comment:
This comment doesn't discuss why having multiple kernel versions installed is a bad thing. If we had multiple kernels installed, and we had the delphix zfs modules installed for each kernel version, then everything would work just fine. It might be worth expanding on this comment to explain the actual failure scenario better.

Reply:
I've updated the comment.

>  It might be worth expanding on this comment to explain the actual failure scenario better.

See 2nd link in description.

Reply:
I think this comment still leaves out the most important information, which is that we can wind up with an image that boots a kernel version that doesn't have the corresponding delphix zfs kernel module package installed, but _does_ have the delphix zfs userspace packages installed.

That is, unless you are trying to convey this in the sentence:

> there are other packages that come with 'linux-generic' that won't be installed for that particular kernel version

?

Reply:
I've updated comments with more details.